### ISSUE TITLE
predefine custom filters for end users

### ISSUE THREAD

**Issue by fredzannarbor** [man-group/dtale#471]
Hi,

I am using dtale (which is great!) to enable authors to deep dive into royalty data so I want to be able to predefine a number of custom filters that are available to these not-always-tech-savvy end users when they log into the standalone dtale app.  For example,  one might be:

`Author.str.contains($authorname) and Format=='Paperback'`

How can I do this?

**Comment by aschonfeld** [comment]
Hmm, this is a good one.  So I could certainly add a hook in the back-end where you could predefine `lambda` functions by column name. For example:
```
dtale.set_predefined_filters({
    'Author': {
        'name': 'Paperback Author',
        'func': lambda v, df: df[df['Author'].str.contains(v) & df['Format'] == 'Paperback'],
        'input': 'text',  # this could be something like 'text' or 'select'
    }
})
```

The tough part will be displaying these filters on the front-end.  Should they be displayed in a separate filter pop-up?  Or should the Column menu for the column in question (Author) display them?  I'll leave that question up to you.

Thanks for the submission

**Comment by fredzannarbor** [comment]
For my end users, I'd go with whatever's most prominent.  A separate popup would probably be good.  

I would definitely use this!

**Comment by aschonfeld** [comment]
@fredzannarbor Just wanted to give you an update on this.  Here's a screen recording of some data loaded using this snippet:
```python
import pandas as pd
import dtale
import dtale.predefined_filters as predefined_filters

predefined_filters.set_filters([
    {
        "name": "A and B > 2",
        "column": "A",
        "description": "Filter A with B greater than 2",
        "handler": lambda df, val: df[(df["A"] == val) & (df["B"] > 2)],
        "input_type": "input",
    },
    {
        "name": "A and (B % 2) == 0",
        "column": "A",
        "description": "Filter A with B mod 2 equals zero (is even)",
        "handler": lambda df, val: df[(df["A"] == val) & (df["B"] % 2 == 0)],
        "input_type": "select",
    },
    {
        "name": "A in values and (B % 2) == 0",
        "column": "A",
        "description": "A is within a group of values and B mod 2 equals zero (is even)",
        "handler": lambda df, val: df[df["A"].isin(val) & (df["B"] % 2 == 0)],
        "input_type": "multiselect",
    }
])

df = pd.DataFrame(
    ([[1, 2, 3, 4, 5, 6], [7, 8, 9, 10, 11, 12], [13, 14, 15, 16, 17, 18]]),
    columns=['A', 'B', 'C', 'D', 'E', 'F']
)
dtale.show(df)
```

https://user-images.githubusercontent.com/11547371/116838322-f85c2100-ab9b-11eb-954d-d2d2bd7dc9e2.mov

**Comment by aschonfeld** [comment]
@fredzannarbor just released v1.45.0 with this new functionality, let me know if you hit any issues
